src/lib/y2network/config_reader/sysconfig.rb

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require "y2network/config"
+require "y2network/interface"
+require "y2network/routing_table"
+
+Yast.import "NetworkInterfaces"
+Yast.import "Routing"
+
+module Y2Network
+  module ConfigReader
+    # This class reads the current configuration from the system
+    class Sysconfig
```

Comment:
I am confused now. This is wicked reader?

Reply:
Wicked reads the sysconfig configuration, and then his own configuration in XML merging all together in his internal state. In YaST we do not deal with the XML configuration but maybe we would like to support that in the future, that is why probably it was suggested to use `Sysconfig` instead of `Wicked`.

Reply:
Yes, that's it. Initially I created a `Wicked` class but @mchf suggested to rename it to `Sysconfig` and reserve the `Wicked` name for future uses (like using the wicked's XML based interface).